### ISSUE TITLE
[Question] SwaggerToCSharpController: Stuck with Swagger to .NET Framework 4.6.x generation

### ISSUE THREAD

**Issue by jawn** [RSuter/NSwag#1716]
TL;DR
Using `SwaggerToCSharpController`, how should the Controller implementation be injected so it is used?
Also, is there an example where this is done with NSwag? I see many examples but couldn't find it or overlooked it.

## Details

I am trying to get the following working:
* Input: swagger 2.0 specification 
* Output: Web API project with working controllers, on .NET Framework 4.6.2 on IIS (Express).

Here's what I did so far

## Step 1: I configured the Web API project for NSwag 

**I followed these instructions** from https://github.com/RSuter/NSwag/wiki/OwinGlobalAsax#integration

![image](https://user-images.githubusercontent.com/1705112/48004981-8f060f00-e112-11e8-9f43-4be1025a4017.png)

## Step 2: **Then added the generated .cs class based on these settings:**

```
{
  "runtime": "Default",
  "defaultVariables": null,
  "swaggerGenerator": {
    "fromSwagger": {
      "json": "swagger: '2.0'***REMOVED***",
      "url": "http://petstore.swagger.io/v2/swagger.json",
      "output": null
    }
  },
  "codeGenerators": {
    "swaggerToCSharpController": {
      "controllerBaseClass": null,
      "controllerStyle": "Partial",
      "useCancellationToken": false,
      "aspNetNamespace": "System.Web.Http",
      "routeNamingStrategy": "None",
      "className": "{controller}",
      "operationGenerationMode": "MultipleClientsFromOperationId",
      "additionalNamespaceUsages": [],
      "additionalContractNamespaceUsages": [],
      "generateOptionalParameters": false,
      "generateJsonMethods": true,
      "enforceFlagEnums": false,
      "parameterArrayType": "System.Collections.Generic.IEnumerable",
      "parameterDictionaryType": "System.Collections.Generic.IDictionary",
      "responseArrayType": "System.Collections.ObjectModel.ObservableCollection",
      "responseDictionaryType": "System.Collections.Generic.Dictionary",
      "wrapResponses": false,
      "wrapResponseMethods": [],
      "generateResponseClasses": true,
      "responseClass": "SwaggerResponse",
      "namespace": "***REMOVED***",
      "requiredPropertiesMustBeDefined": true,
      "dateType": "System.DateTime",
      "jsonConverters": null,
      "dateTimeType": "System.DateTime",
      "timeType": "System.TimeSpan",
      "timeSpanType": "System.TimeSpan",
      "arrayType": "System.Collections.Generic.List",
      "arrayInstanceType": null,
      "dictionaryType": "System.Collections.Generic.Dictionary",
      "dictionaryInstanceType": null,
      "arrayBaseType": "System.Collections.ObjectModel.ObservableCollection",
      "dictionaryBaseType": "System.Collections.Generic.Dictionary",
      "classStyle": "Inpc",
      "generateDefaultValues": true,
      "generateDataAnnotations": true,
      "excludedTypeNames": [],
      "handleReferences": false,
      "generateImmutableArrayProperties": false,
      "generateImmutableDictionaryProperties": false,
      "jsonSerializerSettingsTransformationMethod": null,
      "templateDirectory": null,
      "typeNameGeneratorType": null,
      "propertyNameGeneratorType": null,
      "enumNameGeneratorType": null,
      "serviceHost": null,
      "serviceSchemes": null,
      "output": "generated.cs"
    }
  }
}

```
The results is that the /swagger documentation displays as expected. 👍 

## Step 3. **Then I implemented the IController class** 

However, when calling my API the controller is not called.

**Comment by RSuter** [comment]
You need to register the IController interface and IController implementation in the DI system of your ASP.NET application...

**Comment by RSuter** [comment]
Feel free to update the docs: 
https://github.com/RSuter/NSwag/wiki/SwaggerToCSharpControllerGenerator

There are also quite some open issues regarding controller generation: 
https://github.com/RSuter/NSwag/projects/4

**Comment by jawn** [comment]
Thanks for teaching me about the project feature. I see it's focused on ASP.NET core though, understandably.

**Comment by RSuter** [comment]
> I see it's focused on ASP.NET core though, understandably.

Yes, but ASP.NET support will not be removed and it also supports ASP.NET Core on full .NET as a migration path.
